### PR TITLE
Add automatic handling of writeWithoutResponse to writeDataCharacteristi...

### PR DESCRIPTION
**Pull request by bbx10**
The Adafruit nRF8001 breakout board needs the writeWithoutResponse property so writeDataCharacteristic has been modified to examine the characteristic properties.

### CODE REVIEW

File: lib/noble-device.js [hunk 1]
```diff
@@ -71,7 +71,9 @@ NobleDevice.prototype.readDataCharacteristic = function(serviceUuid, characteris
 };
 
 NobleDevice.prototype.writeDataCharacteristic = function(serviceUuid, characteristicUuid, data, callback) {
-  this._characteristics[serviceUuid][characteristicUuid].write(data, false, function(error) {
+  var c = this._characteristics[serviceUuid][characteristicUuid];
+  var withoutResponse = c.properties.indexOf('writeWithoutResponse') !== -1;
```
Comment (sandeepmistry) [inline lib/noble-device.js]: What do you think about changing this line to:

```javascript
var withoutResponse = (c.properties.indexOf('writeWithoutResponse') !== -1) && (c.properties.indexOf('write') === -1) ;
```

to prefer regular writes for characteristics that have both ```write``` and ```writeWithoutResponse``` properties.

**Comment by sandeepmistry** [comment]
@bbx10 thanks for the suggestion/pull request, please see my comment above, other than that it looks good to merge.

PS: also checkout [arduino-BLEPeripheral](https://github.com/sandeepmistry/arduino-BLEPeripheral) if you're interested in custom services/characteristics with your Adafruit nRF8001.

File: lib/noble-device.js (hunk 1 above)
Comment (bbx10) [inline lib/noble-device.js]: Good point! I did not consider the case of both properties being present. Preferring write in this case makes sense. I will make the chang.

**Comment by bbx10** [comment]
Thanks, I will take a look. My next step is to use noble-device to create a simple nrfuart object that looks like a serial port. This should make it easy for Arduino and node.js programs to communicate.

The step after that is firmata over nrfuart. Adafruit has firmata on their side so I only need to bridge between node.js firmata and nrfuart.

**Comment by sandeepmistry** [comment]
@bbx10 thanks again!

Please let me know when you has format over nrfuart working ...

**Comment by bbx10** [comment]
@sandeepmistry thanks for merging.

I pushed up the first draft of the node nRF8001 driver to [bbx10/node-nRF8001](https://github.com/bbx10/node-nRF8001.git). The README is slightly out of date because I did not see your merge email before I pushed. node firmata works with a forked version of Adafruit_BLEFirmata. The README has the details on making this work.

**Comment by sandeepmistry** [comment]
@bbx10 thanks for sharing.

One note, if you run ```npm install sandeepmistry/noble-device --save``` in your ```node-nRF8001``` folder checkout, it'll update your ```package.json``` to use the master version of ```noble-device```. This should save you from the git clone steps.

I'll publish a new version once your happy with master and your library is ready to be published.

**Comment by bbx10** [comment]
@sandeepmistry 

Thanks for the tip! I am new to npm and javascript so I appreciate the help.

If the other pull request is OK, I will make one more change then re-test. If everything goes OK, I wll be ready to publish.

I am pleasantly surprised to see how little code I had to write when using noble-device. noble-device is a big win for node BLE developers.

I found discoverAll is the first step in raising an army of nodebot minions. :) Well, so far I only have 1 nRF8001 and 1 BLE nano but it's a start. Echo test works with both at the same time.

**Pull request merged**